website_sale_stock_available/models/product.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.multi
+    def _get_combination_info(self, combination=False, product_id=False,
+                              add_qty=1, pricelist=False,
+                              parent_combination=False, only_template=False):
+        template = self.with_context(website_sale_stock_available=True)
+        return super(ProductTemplate, template)._get_combination_info(
+            combination, product_id, add_qty, pricelist, parent_combination,
+            only_template)
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    def _compute_quantities_dict(self, lot_id, owner_id, package_id,
+                                 from_date=False, to_date=False):
+        res = super()._compute_quantities_dict(
+            lot_id, owner_id, package_id, from_date, to_date)
+        if 'website_sale_stock_available' in self.env.context:
+            new_dict = dict(self.env.context)
+            new_dict.pop("website_sale_stock_available", None)
+            for product in self.with_context(new_dict):
+                immediately = product.immediately_usable_qty
+                res[product.id]['virtual_available'] = immediately
```

Comment:
This should be the same:

```suggestion
        for product in self.with_context(website_sale_stock_available=False):
            immediately = product.immediately_usable_qty
            res[product.id]['virtual_available'] = immediately
```

Reply:
I passed website_sale_stock_available=False to context instead of removing it from context, as you stated, but I didn't remove the 'if' statement. I don't want that code block inside the 'if' to be executed if the context doesn't have that variable or it is False.